src/diffusers/models/attention_processor.py

```diff
@@ -2087,43 +2087,55 @@ def __call__(self, attn: Attention, hidden_states: torch.FloatTensor, *args, **k
 
 class IPAdapterAttnProcessor(nn.Module):
     r"""
-    Attention processor for IP-Adapater.
+    Attention processor for Multiple IP-Adapater.
 
     Args:
         hidden_size (`int`):
             The hidden size of the attention layer.
         cross_attention_dim (`int`):
             The number of channels in the `encoder_hidden_states`.
-        num_tokens (`int`, defaults to 4):
+        num_tokens (List[`int`], defaults to [4]):
             The context length of the image features.
-        scale (`float`, defaults to 1.0):
+        scale (`float`, defaults to [1.0]):
             the weight scale of image prompt.
     """
 
-    def __init__(self, hidden_size, cross_attention_dim=None, num_tokens=4, scale=1.0):
+    def __init__(self, hidden_size, cross_attention_dim=None, num_tokens=[4], scale=1.0):
```

Comment:
```suggestion
    def __init__(self, hidden_size, cross_attention_dim=None, num_tokens=(4,), scale=1.0):
```

Can we try to make this an immutable object instead. Not a big fan of having mutable signature defaults. See: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
ahhh I have noticed we have always used tuples but did not know why until now. make sense!